src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SVFileUtils.java

```diff
@@ -0,0 +1,184 @@
+package org.broadinstitute.hellbender.tools.spark.sv.utils;
+
+import htsjdk.samtools.*;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.utils.HopscotchSet;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+import java.io.*;
+import java.util.*;
+
+public final class SVFileUtils {
+
+    private static final String REFERENCE_GAP_INTERVAL_FILE_COMMENT_LINE_PROMPT = "#";
+
+    /**
+     * Creates a directory, in local FS, HDFS, or Google buckets to write to.
+     */
+    public static boolean createDirToWriteTo(final String pathString) {
+        try {
+            Utils.nonNull(pathString);
+            if ( java.nio.file.Files.exists(java.nio.file.Paths.get(pathString)) )
+                throw new IOException("Directory to be created already exists: " + pathString);
+
+            final boolean isSuccessful;
+            if (BucketUtils.isHadoopUrl(pathString)) {
+                isSuccessful = org.apache.hadoop.fs.FileSystem.get(new org.apache.hadoop.conf.Configuration()).mkdirs(new org.apache.hadoop.fs.Path(pathString));
+            } else {
+                final java.nio.file.Path dir = java.nio.file.Files.createDirectory(IOUtils.getPath(pathString));
+                isSuccessful = java.nio.file.Files.isDirectory(dir) && java.nio.file.Files.isWritable(dir);
+            }
+            return isSuccessful;
+        } catch (final IOException x) {
+            throw new GATKException("Could not create directory: " + pathString, x);
+        }
+    }
+
+    public static void writeLinesToSingleFile(final Iterator<String> linesToWrite, final String fileName) {
```

Comment:
Cannot the code using this be changed to `Files.write(Path, Iterable<? extends CharSequence>, OpenOption...)`. If the `Path` is obtained with `IOUtils`, then it should be fine if it is in a bucket too...

Reply:
I still thing that the usage of this method can be substituted by `Files.write(IOUtils.getPath(fileName), linesToWrite)`; there are also utils already implemented to convert between iterable, iterators and stream (see `org.broadinstitute.hellbender.utils.Utils`).

Reply:
@magicDGS Thanks! I've replaced this method with the one you suggested and it is working well.